### PR TITLE
feat: question filters

### PR DESCRIPTION
**Pull request by mariojsnunes**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Followup on https://github.com/ONEARMY/community-platform/pull/3121, #3222

Queries firebase directly for fetching questions, with pagination.
It's possible to filter, sort, and load more items.

This is the new approach, ready for server-side rendering, which doesn't use mobx/fuse.js.
Items are always requested from the database on each search/filter.
Fetches only 10 (configurable) items at a time, instead of the whole collection.

Using the new, modular API from firestore. https://firebase.google.com/docs/firestore/query-data/queries#or_queries
(required for `or` queries)

Deprecated Random filter as they are not possible with firestore queries.
Search no longer supports fuzziness, but I don't think it's that useful or a blocker.

The required indexes are defined on a firebase config file and automatically deployed.

## Git Issues

Closes https://github.com/ONEARMY/community-platform/issues/2968
Solves https://github.com/ONEARMY/community-platform/issues/3307 for Questions Module

### CODE REVIEW

File: src/stores/Question/question.store.tsx [hunk 1]
```diff
@@ -169,12 +170,14 @@ export class QuestionStore extends ModuleStore {
...
     const user = this.activeUser as IUser
     const creatorCountry = this.getCreatorCountry(user, values)
+    const keywords = getKeywords(values.title + ' ' + values.description)
 
     await dbRef.set({
       ...(values as any),
       creatorCountry,
       _createdBy: values._createdBy ?? this.activeUser?.userName,
       slug,
+      keywords: Array.from(keywords),
```
Comment (benfurber) [inline src/stores/Question/question.store.tsx:180]: The tests should be updated to cover this.

File: src/utils/searchHelper.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { stopwords } from './stopwords'
```
Comment (benfurber) [inline src/utils/searchHelper.ts:1]: Would love a little test or two for this util please. Perfect low hanging fruit to keep the coverage OK.

File: src/utils/stopwords.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
+export const stopwords = new Set([
```
Comment (benfurber) [inline src/utils/stopwords.ts:1]: I assume this list of words came from somewhere? Worth referencing in a comment in the file?

**Review by benfurber — CHANGES_REQUESTED** [review]
That service is so nice to read @mariojsnunes. Love all this. And the questions load so much faster!

File: src/pages/Question/QuestionListing.tsx [hunk 4]
```diff
@@ -1,23 +1,84 @@
...
+      }
+      setSearchParams(params)
+    } else {
+      // search only when sort is set (avoids duplicate requests)
+      fetchQuestions()
+    }
+  }, [q, category, sort])
+
+  const fetchQuestions = async (
+    skipFrom?: QueryDocumentSnapshot<DocumentData, DocumentData>,
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply src/pages/Question/QuestionListing.tsx:53]: Fair point, ideally firestore would support passing the item id, but I could only make it work passing the document snapshot. Maybe we could store it in memory at the service, but that could lead to other issues. I think this is the best option for now.

File: src/stores/Question/question.store.tsx (hunk 1 above)
Comment (mariojsnunes) [reply src/stores/Question/question.store.tsx]: Can we do this on another PR? For SSR, this store will have to be removed. I could work on that next.

File: src/pages/Question/question.service.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import {
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply src/pages/Question/question.service.ts:1]: done!

File: src/utils/searchHelper.ts (hunk 2 above)
Comment (mariojsnunes) [reply src/utils/searchHelper.ts:1]: done!

**Comment by mariojsnunes** [comment]
I've added ts-jest package to make it easier to run tests locally

File: src/pages/Question/labels.ts [hunk 6]
```diff
@@ -7,8 +7,9 @@ export const buttons = {
 }
 
 export const headings = {
-  create: 'Ask your question to the community',
-  edit: 'Edit your question to the community',
+  create: 'Ask a question',
+  edit: 'Edit your question',
+  list: 'Ask your questions and help others out',
```
Comment (benfurber) [inline src/pages/Question/labels.ts:12]: This undoes some copy changes I made based on user feedback.

Comment (benfurber) [reply src/pages/Question/labels.ts:12]: If there's only little things like this that need changing, I hope you don't mind me going to the branch myself so I can get this all merged in asap.

**Review by benfurber — APPROVED** [review]

File: src/pages/Question/labels.ts (hunk 6 above)
Comment (mariojsnunes) [reply src/pages/Question/labels.ts]: not at all, please go ahead

**Pull request merged**